enclave-runner/src/usercalls/mod.rs

```diff
@@ -446,13 +674,32 @@ impl EnclaveState {
     ) -> StdResult<(u64, u64), failure::Error> {
         // There is no other way than `Self::library` to get an `Arc<Self>`
         let library = enclave.kind.as_library().unwrap();
-
         let thread = library.threads.lock().unwrap().recv().unwrap();
-        match RunningTcs::entry(
-            enclave.clone(),
-            thread,
-            EnclaveEntry::Library { p1, p2, p3, p4, p5 },
-        ) {
+
+        let (io_queue_send, io_queue_receive) = mpsc::channel();
+        let (work_sender, work_receiver) = crossbeam::crossbeam_channel::unbounded();
+        if work_sender.send(Work::Stopped(thread, EnclaveEntry::Library {p1, p2, p3, p4, p5})).is_err()
+        {
+            panic!("Error sending data via the work sender");
+        }
+
+        // As currently we are not supporting spawning threads let the number of threads be 2
+        // one for usercall handling the other for actually runnign
+        let num_of_threads = 2;
+
+        let result = crossbeam::scope(|scope| {
+            EnclaveState::create_worker_threads(scope, num_of_threads, work_receiver,io_queue_send, enclave.clone());
+            // ///////////////////////////////
+            // // main syscall polling loop //
+            // ///////////////////////////////
+            return EnclaveState::syscall_loop(io_queue_receive, work_sender);
```

Comment:
I don't think it's correct that there's a separate event loop for every library thread. Is this one of the solutions we discussed?

Reply:
Yes. We discussed various things. This was one of the solutions.